pkg/endpointmanager/conntrack.go

```diff
@@ -152,6 +153,51 @@ func ResetProxyPort(ipv4Enabled bool, e *endpoint.Endpoint, isLocal bool, ips []
 	}
 }
 
+type epsInfo struct {
+	keys []lxcmap.EndpointKey
+}
+
+func (eps *epsInfo) GetBPFKeys() []lxcmap.EndpointKey {
+	return eps.keys
+}
+
+func (eps *epsInfo) GetBPFValue() (*lxcmap.EndpointInfo, error) {
+	return nil, nil
+}
+
+// filterEndpointIDs gets the list of endpoints to preserve
+// and the epsInfo object from which all the other endpoints are removed.
+// After execution of that function, epsInfo should contain only endpoints
+// provided in the list.
+func filterEndpointIDs(eps []*endpoint.Endpoint, epsi *epsInfo) {
+	for _, ep := range eps {
+		for _, keyToPreserve := range ep.GetBPFKeys() {
+			for i, key := range epsi.keys {
+				if key == keyToPreserve {
+					epsi.keys = append(epsi.keys[:i], epsi.keys[i+1:]...)
```

Comment:
This will delete all endpoints that do not belong to the list of endpoints running